src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -137,6 +141,17 @@ internal class OnboardingSupervisor(
         }
     }
 
+    private fun retrieveChainRpcEndpoints() {
+        val url = "${helper.deploymentUri}/configs/rpc.json"
+        helper.get(url) { _, response, _, _ ->
+            if (response != null) {
+                Json.decodeFromString<Map<String, RpcInfo>>(response).let {
+                    RpcConfigs.chainIdToRpcMap = it
```

Comment:
There can be a race-condition where the Skip endpoint returns before the config.json returns, in which the case the rcp will be null.  Might want to do the Skip resource fetch after this instead of in parallel.

Reply:
if we do lookups at usage time instead of at processing time, i think we're mostly ok here?

sidenote: we really need to start using some proper async primitives (coroutines or flows or _something_) =/

Reply:
> mostly ok here
yeah unless github.com is extreeeeemely slow.

> we really need to start using some proper async primitives
see latest commit https://github.com/dydxprotocol/v4-abacus/pull/512/commits/20ed7a28620821323577db5f188bde85947eea73